src/test/regress/expected/fkeys_between_local_ref.out

```diff
@@ -0,0 +1,144 @@
+  CREATE TABLE local_table_5 (col_1 INT UNIQUE);
+  -- now define foreign key from local to citus local table
+  ALTER TABLE local_table_5 ADD CONSTRAINT fkey_12 FOREIGN KEY (col_1) REFERENCES local_table_2(col_1);
+  -- now we have 5 citus local tables in this schema
+  SELECT COUNT(*)=5 FROM citus_local_tables_in_schema;
+ ?column?
+---------------------------------------------------------------------
+ t
+(1 row)
+
+ROLLBACK;
+-- they fail as local_table_99 does not exist
+ALTER TABLE local_table_99 ADD CONSTRAINT fkey FOREIGN KEY (col_1) REFERENCES reference_table_1(col_1);
+ERROR:  relation "local_table_99" does not exist
+ALTER TABLE local_table_1 ADD CONSTRAINT fkey FOREIGN KEY (col_1) REFERENCES local_table_99(col_1);
+ERROR:  relation "local_table_99" does not exist
+-- they fail as col_99 does not exist
+ALTER TABLE local_table_1 ADD CONSTRAINT fkey FOREIGN KEY (col_99) REFERENCES reference_table_1(col_1);
+ERROR:  column "col_99" referenced in foreign key constraint does not exist
+ALTER TABLE local_table_1 ADD CONSTRAINT fkey FOREIGN KEY (col_1) REFERENCES reference_table_1(col_99);
+ERROR:  column "col_99" referenced in foreign key constraint does not exist
+-- fails as col_2 does not have a unique/primary key constraint
+ALTER TABLE local_table_1 ADD CONSTRAINT fkey FOREIGN KEY (col_1) REFERENCES reference_table_1(col_2);
+ERROR:  there is no unique constraint matching given keys for referenced table "reference_table_1"
+CREATE TABLE reference_table_2 (col_1 INT UNIQUE, col_2 INT);
+INSERT INTO reference_table_2 SELECT i FROM generate_series(195, 205) i;
+BEGIN;
+  -- define foreign key when both ends are local tables
+  ALTER TABLE local_table_1 ADD CONSTRAINT fkey_11 FOREIGN KEY (col_1) REFERENCES reference_table_2(col_1);
+  -- we still don't convert local tables to citus local tables when
+  -- creating reference tables
+  SELECT create_reference_table('reference_table_2');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+  -- now defining another foreign key would convert local tables to
+  -- citus local tables as reference_table_2 is now a reference table
+  ALTER TABLE local_table_1 ADD CONSTRAINT fkey_12 FOREIGN KEY (col_1) REFERENCES reference_table_2(col_1);
+  -- now print metadata to show that everyting is fine
+  SELECT logicalrelid::text AS tablename, partmethod, repmodel FROM pg_dist_partition
+  WHERE logicalrelid::text IN (SELECT tablename FROM pg_tables WHERE schemaname='fkeys_between_local_ref')
+  ORDER BY tablename;
+     tablename     | partmethod | repmodel
+---------------------------------------------------------------------
+ local_table_1     | n          | c
+ local_table_2     | n          | c
+ local_table_3     | n          | c
+ local_table_4     | n          | c
+ reference_table_1 | n          | t
+ reference_table_2 | n          | t
+(6 rows)
+
```

Comment:
noting here not to forget:

let's add some tests for unsupported fkeys between citus local tables & reference tables (like ON UPDATE CASCADE)

Reply:
yes, let's have one test like that (or data doesn't good to create fkey)

Reply:
ping on this

Reply:
14b8b5e6ba48ed367d3aff15fd66cdec2c4bafd5